### PR TITLE
Use IPv6 on server in multiple H/2 connection tests

### PR DESCRIPTION
**Pull request by alnikola**

### CODE REVIEW

File: src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs [hunk 1]
```diff
@@ -2077,12 +2077,18 @@ public sealed class SocketsHttpHandlerTest_Http2 : HttpClientHandlerTest_Http2
     {
         public SocketsHttpHandlerTest_Http2(ITestOutputHelper output) : base(output) { }
 
-        [ConditionalFact(nameof(SupportsAlpn))]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/41078")]
-        public async Task Http2_MultipleConnectionsEnabled_ConnectionLimitNotReached_ConcurrentRequestsSuccessfullyHandled()
+        public static IEnumerable<object[]> LongRunning()
         {
+            return Enumerable.Repeat(true, 10000).Select((b, i) => new object[] { i % 2 == 0 }).ToArray();
```
Comment (antonfirsov) [inline src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs:2082]: I think Xunit will collapse repeated parameters here, reducing your theories to 2 cases. ( `[true,false,true,false,true,false] => [true,false]` )

Comment (alnikola) [reply src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs:2082]: Hmm, why do you think so? Locally, it runs 10000 iterations as expected and I saw many runs on the CI result page (though, I didn't count them).

Comment (antonfirsov) [reply src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs:2082]: The following test reports only 2 test cases on `dotnet test`, with `1` and `2` in error messages:

```C#
using System;
using Xunit;
using System.Linq;
using System.Collections.Generic;
using System.Threading;

namespace XunitRepeatedParameters
{
    public class UnitTest1
    {
    	public static IEnumerable<object[]> Stuff => new object[][]{
    		new object[]{ true },
    		new object[]{ false },
    		new object[]{ true },
    		new object[]{ false },
    		new object[]{ true },
    		new object[]{ false },
    		new object[]{ true },
    		new object[]{ false },
    	};

    	private static int _counter = 0;

        [Theory]
        [MemberData(nameof(Stuff))]
        public void Test1(bool b)
        {
        	Interlocked.Increment(ref _counter);
        	throw new Exception($"{_counter	}");
        }
    }
}

```

There might be some difference in `Http.Tests` I'm not considering though.

Comment (alnikola) [reply src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs:2082]: > The following test reports only 2 test cases on dotnet test, with 1 and 2 in error messages:

Where do you see that there are only 2 test cases? Locally, in the console or somewhere on CI? Could you please post either log or link?

Comment (antonfirsov) [reply src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs:2082]: Very strange, adding this to `System.Net.Http` `FunctionalTests` and running it there behaves differently than a console project from scratch. Nevermind me then if it works there.

**Pull request closed without merging**